### ISSUE TITLE
Installation Issues

### ISSUE THREAD

**Issue by rizar** [bartvm/blocks#173]
My attempt to run blocks at another machine ended up lie this:

```
╰─$ python setup.py -h
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    import blocks
  File "/home/rizar/Dist/blocks/blocks/__init__.py", line 2, in <module>
    from blocks.config_parser import config  # flake8: noqa
  File "/home/rizar/Dist/blocks/blocks/config_parser.py", line 45, in <module>
    import yaml
ImportError: No module named yaml
```

**Comment by rizar** [comment]
OK, I see, it fetches version: but this should be done without loading any third-party modules.

Another issue: we should fix minimal version of dependencies, because e.g. six 1.2 does not have `add_metaclass`.

**Comment by rizar** [comment]
Another thing: shouldn't live plotting be made optional for the MNIST example? It is not nice that right after installation of Blocks I can not run it because I do not have bokeh..

**Comment by bartvm** [comment]
Ah, I think you're right, forgot to add `yaml` to the dependencies (it's a dependency of Pylearn2, and a very common dependency in general, so I guess that's why it never showed up in any other system).

Agreed on not making plotting part of the standard example.

**Comment by bartvm** [comment]
What system were you using!? Six 1.2 is from 2013, 2.5 years old...

**Comment by bartvm** [comment]
The YAML and version issue were fixed in 2aa45e7844f896e9b28fe2dbb7e5b40cb2bcfb3a

**Comment by rizar** [comment]
Thanks!

I guess it was Ubuntu 12.04 or smth like that :)
